### PR TITLE
chore(Cross): [IOAPPX-435] Improve overall UI typography + Update `io-app-design-system` to `4.0.1`

### PR DESCRIPTION
**Pull request by dmnplb**
## Short description
This PR improves overall UI typography.

## List of changes proposed in this pull request
- Update `io-app-design-system` to the new major `4.0.1` version
- Update font weight in the `HeaderFirstLevel` to make the title bolder
- Improve **Strong** markdown style by replacing `Bold` with the `Semibold` style
- Decrease font weight of the subtitle in the `ServicesHeader` component
- Improve formatting of the `MessageListItem`
- Replace removed `Chip` typographic style with `LabelMini`
- Replace removed `Label` typographic style with `Body`
- Replace removed `LabelSmall` typographic style with the new `BodySmall`
- **[extra]** Fix typo in the English locale (`CIEIdWizard`)

> [!note]
> **For the reviewers:** the default weight of `LabelSmall` was `Semibold`, while the default weight for `BodySmall` is `Regular`. That's why `LabelSmall` set without the `weight` prop was replaced by `BodySmall` with the `weight` prop  set to `Semibold`. The same should happen to the `Label` typographic style

### Preview
| Actual UI | New UI |
|--------|--------|
| Simulator Screenshot - iPhone 16 Pro - 2024-11-21 at 11 47 43 | Simulator Screenshot - iPhone 16 Pro - 2024-11-21 at 11 52 44 | 

#### Old vs New

<details><summary><code>Messages Home</code></summary>
<p>

| Before | After |
|--------|--------|
| Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 19 58 | Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 22 33 | 

</p>
</details> 

<details><summary><code>Payments Home</code></summary>
<p>

| Before | After |
|--------|--------|
| Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 20 22 | Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 23 51 | 

</p>
</details> 

### Related PRs
* https://github.com/pagopa/io-app-design-system/pull/360
* https://github.com/pagopa/io-app-design-system/pull/361

## How to test
Check each DS component and navigate through the app to check if everything is OK

### CODE REVIEW

**Review by LeleDallas — APPROVED** [review]
Great job! B&P sections seems good!

**Comment by shadowsheep1** [comment]
@dmnplb Can we add in this PR description some Android screenshots too? And what about a11y? Do you think that this typographic changes could impact a11y too or they won't impact it?

File: ts/components/IOMarkdown/renderRules.tsx [hunk 1]
```diff
@@ -160,7 +159,7 @@ export const DEFAULT_RULES: IOMarkdownRenderRules = {
    */
   Strong(strong: TxtStrongNode, render: Renderer) {
     return (
-      <Text key={getTxtNodeKey(strong)} style={{ fontWeight: "800" }}>
```
Comment (Ladirico) [inline ts/components/IOMarkdown/renderRules.tsx:163]: I've a doubt: Why you have change this fontWeight from 800 to 600? Because the "bold" size is no longer supported?

File: ts/features/bonus/cgn/components/CgnCard.tsx [hunk 2]
```diff
@@ -61,15 +61,19 @@ export const CgnCard = ({ expireDate, withEycaLogo }: CgnCardProps) => {
...
+        <BodySmall
+          weight="Semibold"
+          color="blueItalia-850"
+          style={{ width: "70%" }}
+        >
           {I18n.t("bonus.cgn.departmentName")}
-        </LabelSmall>
-        <LabelSmall color="blueItalia-850">
+        </BodySmall>
+        <BodySmall weight="Semibold" color="blueItalia-850">
```
Comment (Ladirico) [inline ts/features/bonus/cgn/components/CgnCard.tsx:71]: Why here you've add `weight="Semibold"`? 🤔

File: ts/features/idpay/code/screens/IdPayCodeDisplayScreen.tsx [hunk 3]
```diff
@@ -21,6 +20,7 @@ import {
   AppParamsList,
   IOStackNavigationProp
 } from "../../../../navigation/params/AppParamsList";
+import useContentWithFF from "../../../../screens/profile/useContentWithFF";
```
Comment (Ladirico) [inline ts/features/idpay/code/screens/IdPayCodeDisplayScreen.tsx:23]: I believe that after the upgrade to branch master it will no longer exist

File: ts/features/itwallet/common/components/ItwSkeumorphicCard/ClaimLabel.tsx [hunk 4]
```diff
@@ -20,8 +14,8 @@ export type ClaimLabelProps = {
 export const ClaimLabel: React.FunctionComponent<
   React.PropsWithChildren<ClaimLabelProps>
 > = ({
-  fontSize = defaultFontSize,
-  fontWeight = defaultFontWeight,
+  fontSize = 11,
+  fontWeight = "Semibold",
```
Comment (Ladirico) [inline ts/features/itwallet/common/components/ItwSkeumorphicCard/ClaimLabel.tsx:18]: why did you remove the variables?

**Review by Ladirico — CHANGES_REQUESTED** [review]

**Comment by dmnplb** [comment]
> @dmnplb Can we add in this PR description some Android screenshots too? And what about a11y? Do you think that this typographic changes could impact a11y too or they won't impact it?

@shadowsheep1 As you can see from [the difference between the actual and proposed DS release](https://github.com/pagopa/io-app-design-system/compare/v3.1.0...v4.0.1#files_bucket), there's no specific change to the layout logic that would justify an in-depth accessibility analysis. So I'm not against a test on a11y, but I think it's a waste of time.

File: ts/features/bonus/cgn/components/CgnCard.tsx (hunk 2 above)
Comment (dmnplb) [reply ts/features/bonus/cgn/components/CgnCard.tsx:71]: @Ladirico The default weight of `LabelSmall` was `Semibold` too, so we apply the same logic applied to the `Body` style

File: ts/features/itwallet/common/components/ItwSkeumorphicCard/ClaimLabel.tsx (hunk 4 above)
Comment (dmnplb) [reply ts/features/itwallet/common/components/ItwSkeumorphicCard/ClaimLabel.tsx:18]: @Ladirico They are not exported, so they are not used externally, and they are only used once. So why declare the variable in the first place? :)

File: ts/components/IOMarkdown/renderRules.tsx (hunk 1 above)
Comment (dmnplb) [reply ts/components/IOMarkdown/renderRules.tsx:163]: @Ladirico The `Bold` font weight is represented by the `700` value, so by setting `800` we are setting an even bolder weight. With the new `Titillio` typeface, this new weight of `800` results in an `ExtraBold` style. To illustrate, it's even more visible than a headline style. So we'll render the `Strong` style with the `Semibold` style because it's more subtle.

Comment (Ladirico) [reply ts/components/IOMarkdown/renderRules.tsx:163]: thanks!

**Review by Ladirico — APPROVED** [review]
LGTM!

**Review by shadowsheep1 — APPROVED** [review]
For A&I Tested both onboarding and first onboarding and all the settings pages on both Android and iOS physical devices. Identification Modal an Block Modal too. All looks good!

**Comment by shadowsheep1** [comment]
> > @dmnplb Can we add in this PR description some Android screenshots too? And what about a11y? Do you think that this typographic changes could impact a11y too or they won't impact it?
> 
> @shadowsheep1 As you can see from [the difference between the actual and proposed DS release](https://github.com/pagopa/io-app-design-system/compare/v3.1.0...v4.0.1#files_bucket), there's no specific change to the layout logic that would justify an in-depth accessibility analysis. So I'm not against a test on a11y, but I think it's a waste of time.

@dmnplb Okay for the a11y. For the Android part instead, I always suggest to add examples of "old" and "new" of the two platforms (Android and iOS).
